hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/virtual/ViewTable.java

```diff
@@ -62,7 +63,7 @@ protected List<TableField> initFields() {
             throw QueryException.error("Cycle detected in view references");
         }
         expansionStack.push(viewPath);
-        SqlNode sqlNode = context.parse(viewQuery).getNode();
+        SqlNode sqlNode = context.parse(viewQuery, NoOpSqlSecurityContext.INSTANCE).getNode();
```

Comment:
if someone manages to create the view (eg. by directly writing to __sql.catalog IMap), such view will be vulnerable.

Reply:
https://github.com/hazelcast/hazelcast/pull/25784/commits/898812b5a218576dfbcafa5b86009cd4c3d53c70